src/metrics/encoding/protobuf/reset_test.go

```diff
@@ -178,6 +178,83 @@ func TestReuseMetricWithMetadatasProtoOnlyCounter(t *testing.T) {
 	require.True(t, cap(input.CounterWithMetadatas.Metadatas.Metadatas) > 0)
 }
 
+func TestReuseMetricForEncoding(t *testing.T) {
+	input := metricpb.MetricWithMetadatas{
```

Comment:
Pretty hard to read this. Consider breaking out the Pipelines definitions, just make it more readable.